cadquery/occ_impl/shapes.py

```diff
@@ -3984,3 +4094,882 @@ def edgesToWires(edges: Iterable[Edge], tol: float = 1e-6) -> List[Wire]:
     ShapeAnalysis_FreeBounds.ConnectEdgesToWires_s(edges_in, tol, False, wires_out)
 
     return [Wire(el) for el in wires_out]
+
+
+#%% utilities
+
+
+def _get(s: Shape, ts: Union[Shapes, Tuple[Shapes, ...]]) -> Iterable[Shape]:
+    """
+    Get desired shapes or raise an error.
+    """
+
+    # convert input into tuple
+    if isinstance(ts, tuple):
+        types = ts
+    else:
+        types = (ts,)
+
+    # validate the underlying shape, compounds are unpacked
+    t = s.ShapeType()
+
+    if t in types:
+        yield s
+    elif t == "Compound":
+        for el in s:
+            if el.ShapeType() in ts:
+                yield el
+            else:
+                raise ValueError(
+                    f"Required type(s): {types}; encountered {el.ShapeType()}"
+                )
+    else:
+        raise ValueError(f"Required type(s): {types}; encountered {t}")
+
+
+def _get_one(s: Shape, ts: Union[Shapes, Tuple[Shapes, ...]]) -> Shape:
+    """
+    Get one shape or raise an error
```

Comment:
docstrings: For consistency add period here?
Get one shape or raise an error
->
Get one shape or raise an error.

Period missing in following four functions as well.